backend/src/controllers/hint.controller.js

```
@@ -0,0 +1,209 @@
+      const { statusCode, payload } = internalServerError(
+        "UPDATE_HINT_ERROR",
+        error.message
+      );
+      res.status(statusCode).json(payload);
+    }
+  }
+
+  async deleteHint(req, res) {
+    const { hintId } = req.params;
+
+    if (isNaN(hintId) || hintId.trim() === "") {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)